lib/galaxy/webapps/galaxy/api/__init__.py

```diff
@@ -198,7 +198,9 @@ def base(self) -> str:
 
     @property
     def host(self) -> str:
-        return str(self.__request.client.host)
+        host = self.__request.client.host
+        assert host is not None
```

Comment:
I think we've tended to use exceptions for this in the past. No strong preference from my side

Reply:
I think we've done that for 4XX errors - these 5XX things maybe less consistently? Assert feels like the right thing to do typing wise? It would be cool to catch all assertions errors and translate them to proper 5XX errors on the way out of the app though.

Reply:
I think @nsoranzo's argument was that asserts can be disabled. This is just to help out the type checker, this would otherwise only happen outside of a web request.